compilation_tests/ctest_layout_type_check.cpp

```diff
@@ -0,0 +1,88 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+#include "ctest_common.hpp"
+
+#include <experimental/mdspan>
+
+namespace stdex = std::experimental;
+
+using E1 = stdex::extents<int32_t, stdex::dynamic_extent, 3>;
+using MLR1 = stdex::layout_right::mapping<E1>;
+// Note we already separately tested that the extents interface is correct
+MDSPAN_STATIC_TEST(
```

Comment:
Optionally, would you consider making the test templated on the layout, so that you don't have to repeat all the conditions three times?  We'll have to add two more times due to P2642.

Reply:
there is other places where we could do that, but I am not sure how to right now. Note that the macro only takes a single statement kinda.